tools/Raven.Migrator/MongoDB/MongoDBMigrator.cs

```diff
@@ -0,0 +1,264 @@
+            }
+        }
+
+        public async Task MigrateDatabse()
+        {
+            AssertDatabaseName();
+
+            var client = CreateNewMongoClient();
+            var database = client.GetDatabase(_configuration.DatabaseName);
+
+            if (_configuration.CollectionsToMigrate == null || 
+                _configuration.CollectionsToMigrate.Count == 0 && _configuration.MigrateGridFS == false)
+            {
+                _configuration.CollectionsToMigrate = await GetCollectionsToMigrate(database);
+            }
+
+            await MigrationHelpers.MigrateNoSqlDatabase(
+                _configuration,
+                async (mongoCollectionName, ravenCollectionName, jsonTextWriter, streamWriter) => 
+                    await MigrateSingleCollection(database, mongoCollectionName, ravenCollectionName, jsonTextWriter),
+                async (jsonTextWriter, streamWriter) =>
+                {
+                    if (_configuration.MigrateGridFS)
+                        await MigrateGridFS(database, jsonTextWriter, streamWriter);
+                });
+        }
+
+        private async Task MigrateSingleCollection(
+            IMongoDatabase database, 
+            string mongoCollectionName,
+            string ravenCollectionName,
+            JsonTextWriter jsonTextWriter)
+        {
+            var collection = database.GetCollection<ExpandoObject>(mongoCollectionName);
+
+            using (var documentsCursor = await collection.Find(_filterDefinition).ToCursorAsync())
+            {
+                while (await documentsCursor.MoveNextAsync())
+                {
+                    foreach (var document in documentsCursor.Current)
+                    {
+                        var dictionary = (IDictionary<string, object>)document;
+                        var documentId = dictionary[MongoDocumentId].ToString();
+
+                        MigrationHelpers.WriteDocument(
+                            document,
+                            documentId,
+                            ravenCollectionName,
+                            _propertiesToRemove,
+                            jsonTextWriter);
+                    }
+                }
+            }
+        }
+
+        private static async Task<List<Collection>> GetCollectionsToMigrate(IMongoDatabase database)
+        {
+            var collectionsToMigrate = await GetCollections(database);
+            return collectionsToMigrate
+                .Select(x => new Collection
+                {
+                    Name = x,
+                    NewName = x
+                }).ToList();
+        }
+
+        private static async Task MigrateGridFS(
+            IMongoDatabase database, 
+            JsonTextWriter jsonTextWriter, 
+            StreamWriter streamWriter)
+        {
+            var bucket = new GridFSBucket(database);
+            var collectionName = bucket.Options.BucketName == "fs" ? "Files" : bucket.Options.BucketName;
```

Comment:
Can we have a casing issue here?

Reply:
The default name for GridFS is "fs".
If it was changed we'll take it as the collection name.